pkg/private/archive.py

```diff
@@ -346,9 +294,9 @@ def add_file(self,
         tarinfo.size = os.fstat(f.fileno()).st_size
         self._addfile(tarinfo, f)
     else:
-      if kind == tarfile.DIRTYPE:
-        self.directories.add(name)
       self._addfile(tarinfo)
+    if kind == tarfile.DIRTYPE:
+      self.directories.add(name.rstrip('/'))
```

Comment:
nit: `name` is not going to end with `/` before reaching this line, so the rstripping seems unnecessary?

Reply:
Yes. add_root promises that name does not end in /.
Changed to an assert.